app/src/main/java/com/andela/gkuti/mipista/service/LocationDetector.java

```diff
@@ -0,0 +1,179 @@
+import android.location.Address;
+import android.location.Geocoder;
+import android.location.Location;
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.support.v4.content.ContextCompat;
+
+import com.google.android.gms.common.ConnectionResult;
+import com.google.android.gms.common.api.GoogleApiClient;
+import com.google.android.gms.location.LocationListener;
+import com.google.android.gms.location.LocationRequest;
+import com.google.android.gms.location.LocationServices;
+
+import java.io.IOException;
+import java.util.List;
+
+/**
+ * LocationDetector class
+ */
+public class LocationDetector implements GoogleApiClient.ConnectionCallbacks, GoogleApiClient.OnConnectionFailedListener, LocationListener {
+    private GoogleApiClient googleApiClient;
+    private Context context;
+    private String country;
+    private String state;
+    private String locality;
+    private String street;
+    private double longitude;
+    private double latitude;
+    private Intent intent;
+    private String newLocation = "";
+
+    /**
+     * Constructor for LocationDetector class
+     *
+     * @param context activity context
+     */
+    public LocationDetector(Context context) {
+        this.context = context;
+        intent = new Intent("location");
+    }
+
+    /**
+     * called when connection was successful
+     */
+    @Override
+    public void onConnected(@Nullable Bundle bundle) {
+        if (ContextCompat.checkSelfPermission(context, Manifest.permission.ACCESS_FINE_LOCATION)
+                == PackageManager.PERMISSION_GRANTED) {
+            LocationRequest locationRequest = LocationRequest.create();
+            locationRequest.setPriority(LocationRequest.PRIORITY_HIGH_ACCURACY);
+            locationRequest.setInterval(60000);
+            LocationServices.FusedLocationApi.requestLocationUpdates(googleApiClient, locationRequest, this);
+        }
+    }
+
+    /**
+     * called when the connection was suspended
+     */
+    @Override
+    public void onConnectionSuspended(int i) {
+    }
+
+    /**
+     * called when the connection failed
+     */
+    @Override
+    public void onConnectionFailed(@NonNull ConnectionResult connectionResult) {
+    }
+
+    /**
+     * method for setting up a connection
+     */
+    public void connect() {
+        country = " ";
+        state = " ";
+        locality = " ";
+        street = " ";
+        googleApiClient = new GoogleApiClient.Builder(context).addApi(LocationServices.API)
+                .addConnectionCallbacks(this)
+                .addOnConnectionFailedListener(this)
+                .build();
+        googleApiClient.connect();
+    }
+
+    /**
+     * method for disconnecting the connection
+     */
+    public void disconnect() {
+        try {
+            LocationServices.FusedLocationApi.removeLocationUpdates(
+                    googleApiClient, this);
+        } catch (Exception e) {
```

Comment:
Resolved: forgot to remove it, I was using it then for different logic on the location detection